ponycheck/generator.pony

```diff
@@ -426,7 +405,7 @@ primitive Generators
               .take(size)
             )
           let shrink_iter: Iterator[Set[T]^] =
-            Iter[USize](CountdownIter(size, 0)) // Range(size, 0, -1))
+            Iter[USize](Range(size-1, -1, -1)) 
```

Comment:
Dang! While i like that PR, i think this will not work as expected. If you provide a literal -1 for a USize, you will get USize.max_value() as -1 is actually out of range for USize.

What this uncovers is a weakness of the stdlib Range in that it cannot count backwards to any number types min_value() as the "upper" bound in range is always exclusive.

Also, first of all actually, negative steps are only possible for signed types as described in the docs: https://stdlib.ponylang.io/collections-Range/

So i think, we either keep the CountDownIter, change the comment, as this one detinitely has the right to exist, given the current state of Range. Or we fix Range first, to both allow negative steps for unsigned types AND allow to specify ranges from and to invlusive and exclusive bounds. Then we can easily exchange the usage of CountdownIter here with Range.

Sorry, that this one has to be so hard!

Reply:
I can definitely do the Range part somewhen until the end of the year.

Reply:
Just to establish a full understanding sth. like:

```
Range[USize](size-1, -1, -1)
```

Will actually not go backwards but go from size-1 to USize.max_value() with a step of USize.max_value(). That is it will already overflow at the first step, and will iterate further and further until by some chance the game of adding USize.max_value() ends up at USize.max_value(), which then stops the Range. This is clearly not intended behaviour and a problem of the api of Range.

Reply:
I see. I didn't think of it but it makes sense. My fix was naive but I was expecting that this inclusion/exclusion thing would have created some problem. We can keep the PR open until the next release of Pony with the fixed Range and retry when it's there.

How would you cover this case with a test?